### PR TITLE
feat(core/presentation): Create IFormFieldApi interface and implement t in FormField and FormikFormField

### PR DESCRIPTION
**Pull request by christopherthielen**
Will be using this in an upcoming PR for managing form sections

### CODE REVIEW

File: app/scripts/modules/core/src/presentation/forms/interface.ts [hunk 1]
```diff
@@ -46,3 +46,12 @@ export interface ICommonFormFieldProps {
   input: React.ComponentType<IFormInputProps>;
   layout?: React.ComponentType<IFieldLayoutProps>;
 }
+
+export interface IFormFieldApi {
```
Comment (christopherthielen) [inline app/scripts/modules/core/src/presentation/forms/interface.ts]: here is the api

File: app/scripts/modules/core/src/presentation/forms/FormikFormField.tsx [hunk 2]
```diff
@@ -56,27 +67,41 @@ export class FormikFormField<T = any> extends React.Component<IFormikFormFieldPr
     }));
   };
 
+  public name = () => this.props.name;
+
+  public label = () => ifString(this.props.label);
```
Comment (jrsquared) [inline app/scripts/modules/core/src/presentation/forms/FormikFormField.tsx]: Isn't `this.props.label` required to be a string?

Comment (christopherthielen) [reply app/scripts/modules/core/src/presentation/forms/FormikFormField.tsx]: It can be any `React.Node` including JSX

**Comment by jrsquared** [comment]
Ahh, got it. Makes sense

**Pull request merged**